support/kind/kind.go

```diff
@@ -28,51 +29,68 @@ import (
 
 var kindVersion = "v0.11.0"
 
-type KindCluster struct {
+type Cluster struct {
 	name string
 	e    *gexe.Echo
 }
 
-func NewKindCluster(name string) *KindCluster {
-	return &KindCluster{name: name, e: gexe.New()}
+func NewKindCluster(name string) *Cluster {
+	return &Cluster{name: name, e: gexe.New()}
 }
 
-func (k *KindCluster) Create() error {
+func (k *Cluster) Create() (string, error) {
 	log.Println("Creating kind cluster ", k.name)
 	// is kind program available
 	if err := findOrInstallKind(k.e); err != nil {
-		return err
+		return "", err
 	}
 
 	if strings.Contains(k.e.Run("kind get clusters"), k.name) {
-		log.Println("Skipping KindCluster.Create: cluster already created: ", k.name)
-		return nil
+		log.Println("Skipping Kind Cluster.Create: cluster already created: ", k.name)
+		return "", nil
 	}
 
 	// create kind cluster using kind-cluster-docker.yaml config file
 	log.Println("launching: kind create cluster --name", k.name)
 	p := k.e.RunProc(fmt.Sprintf(`kind create cluster --name %s`, k.name))
 	if p.Err() != nil {
-		return fmt.Errorf("failed to create kind cluster: %s : %s", p.Err(), p.Result())
+		return "", fmt.Errorf("failed to create kind cluster: %s : %s", p.Err(), p.Result())
 	}
 
 	clusters := k.e.Run("kind get clusters")
 	log.Println("kind clusters available: ", clusters)
 
-	return nil
+	// grab kubeconfig file for cluster
+	kubecfg := fmt.Sprintf("%s-kubecfg", k.name)
+	p = k.e.StartProc(fmt.Sprintf(`kind get kubeconfig --name %s`, k.name))
+	if p.Err() != nil {
+		return "", fmt.Errorf("kind get kubeconfig: %s: %w", p.Result(), p.Err())
+	}
+
+	file, err := ioutil.TempFile("", fmt.Sprintf("kind-cluser-%s", kubecfg))
+	if err != nil {
+		return "", fmt.Errorf("kind kubeconfig file: %w", err)
+	}
+
+	defer file.Close()
+	if n, err := io.Copy(file, p.Out()); n == 0 || err != nil {
+		return "", fmt.Errorf("kind kubecfg file: bytes copied: %d: %w]", n, err)
+	}
+
+	return file.Name(), nil
 }
 
-func (k *KindCluster) GetKubeConfig() (io.Reader, error) {
+func (k *Cluster) GetKubeConfig() (io.Reader, error) {
```

Comment:
If you gonna use the code starting on line 65. Then you don't need that method, no ?

Reply:
Yes, this function is not required.